### PR TITLE
build(examples,templates): refactor build scripts

### PR DESCRIPTION
**Pull request by dk1a**
fixes https://github.com/latticexyz/mud/issues/997

### CODE REVIEW

File: examples/minimal/packages/contracts/package.json [hunk 1]
```diff
@@ -4,20 +4,19 @@
   "private": true,
   "license": "MIT",
   "scripts": {
-    "build": "forge clean && forge build",
-    "deploy": "pnpm run initialize && mud deploy",
-    "deploy:testnet": "pnpm run initialize && mud deploy --profile=lattice-testnet",
+    "build": "pnpm run build:mud && pnpm run build:abi && pnpm run build:types",
+    "build:abi": "forge clean && forge build",
+    "build:mud": "mud tablegen && mud worldgen && mud tsgen --configPath mud.config.ts --out ../client-vanilla/src/mud && mud tsgen --configPath mud.config.ts --out ../client-react/src/mud && mud tsgen --configPath mud.config.ts --out ../client-phaser/src/mud",
+    "build:types": "rimraf types && typechain --target=ethers-v5 out/IWorld.sol/IWorld.json",
```
Comment (holic) [inline examples/minimal/packages/contracts/package.json:10]: we call this step `build:typechain` in other places, can we name that similarly here?

File: templates/phaser/packages/contracts/package.json [hunk 2]
```diff
@@ -4,20 +4,18 @@
...
-    "build": "forge clean && forge build",
-    "deploy": "pnpm run initialize && mud deploy",
-    "deploy:hackathon": "pnpm run initialize && mud deploy --profile=hackathon-testnet",
-    "deploy:testnet": "pnpm run initialize && mud deploy --profile=lattice-testnet",
+    "build": "pnpm run build:mud && pnpm run build:abi && pnpm run build:types",
+    "build:abi": "forge clean && forge build",
+    "build:mud": "mud tablegen && mud worldgen && mud tsgen --configPath mud.config.ts --out ../client/src/mud",
+    "build:types": "rimraf types && typechain --target=ethers-v5 out/IWorld.sol/IWorld.json",
+    "deploy": "pnpm run build && mud deploy",
+    "deploy:hackathon": "pnpm run build && mud deploy --profile=hackathon-testnet",
```
Comment (holic) [inline templates/phaser/packages/contracts/package.json:12]: I think we can prob get rid of this now?

**Review by holic — COMMENTED** [review]
yesss, thank youuu for cleaning this up! this has been bugging me

File: templates/phaser/packages/contracts/package.json (hunk 2 above)
Comment (dk1a) [reply templates/phaser/packages/contracts/package.json]: also removed the profiles from foundry.toml

**Comment by dk1a** [comment]
not sure what's happening with playwright deps, will look into it later

**Review by holic — APPROVED** [review]

**Comment by holic** [comment]
(merging this because I need it to fix CI issues in other PRs)

**Comment by dk1a** [comment]
> (merging this because I need it to fix CI issues in other PRs)

tbf I'm not sure this fixes everything, an incorrect binary cache could be the new source of rare flakiness. Anyways, merging it and waiting for it to trigger is easiest

**Pull request merged**